### PR TITLE
[21803] Improve info displayed when there is no data for a given resource

### PR DESCRIPTION
**Pull request by leewaa**
Work package: [21803](https://community.openproject.org/work_packages/21803)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://leewaa-21798-no-results-info.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://leewaa-21798-no-results-info.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by leewaa** [comment]
My part is done, @mk84 your turn, when ever you're ready.

**Comment by mk84** [comment]
#### Before

op_before

#### After

op_after

File: app/helpers/no_results_helper.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  # form with only the title. Providing an action_url and display_action: true
+  # Displays the box with the title and link to the passed in url.
+  # The title and action_text are found using the locales key lookup unless
+  # custom_title and custom_action_text are provided.
+  def no_results_box(action_url:         nil,
+                     display_action:     false,
+                     custom_title:       nil,
+                     custom_action_text: nil)
+
+    title = custom_title || t('.no_results_title_text', cascade: true) || ''
```
Comment (oliverguenther) [inline app/helpers/no_results_helper.rb]: Why did you add the last `|| ''` there? I don't think it will ever be reached, since `t` would return a missing translation error, wouldn't it?

**Comment by oliverguenther** [comment]
Visual and code changes LGTM. :+1: There seems to be a legitimate cuke failing. Once that's fixed, this is ready to merge from my side.

File: app/helpers/no_results_helper.rb (hunk 1 above)
Comment (leewaa) [inline app/helpers/no_results_helper.rb]: Ah yeah you're right :)

**Comment by leewaa** [comment]
Ok cool, I'll fix that spec, and rebase, then should be good to merge

**Comment by leewaa** [comment]
The remaining failing specs are coming from dev. @oliverguenther good to merge

**Comment by oliverguenther** [comment]
Did you have any specification on what to do with the plugins that also use `:label_no_data` ? A quick search comes up with hits in the following plugins:
- openproject-backlogs
- openproject-costs
- openproject-documents
- openproject-meeting

It would probably be best to have them use the new helper. Also, does this PR break anything in the legacy no data notification?

**Comment by leewaa** [comment]
1. I can just go through the plugins and replace any `no_data` containers with the `no_results` helper where applicable. 
2. Not that I'm aware of.

**Comment by lindenthal** [comment]
Changing the plugins is out of scope for 5.0.0.

**Comment by oliverguenther** [comment]
@lindenthal Thanks for the clarification. I'm merging this then, since it does not affect the plugins in any way other than introducing a styling inconsistency.

I've noted this inconsistency in the following work package and scheduled it for 5.1: https://community.openproject.org/projects/openproject/work_packages/21850
(Please move to another version according to your scope).
